arrow-core-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -43,7 +55,26 @@ object FoldableLaws {
       Law("Foldable Laws: firstOrNone returns None if predicate fails") { FF.firstOrNoneReturnsNoneIfPredicateFails(GEN) },
       Law("Foldable Laws: firstOrNone is consistent with find") { FF.`firstOrNone is consistent with find`(GEN, EQOptionInt) },
       Law("Foldable Laws: firstOrNone is consistent with find matching predicate") { FF.`firstOrNone is consistent with find predicate`(GEN, EQOptionInt) },
-      Law("Foldable Laws: toList turn items into a list") { FF.`toList turn items into a list`(GEN, EQ) }
+      Law("Foldable Laws: toList turn items into a list") { FF.`toList turn items into a list`(GEN, EQ) },
+      Law("Foldable Laws: fold returns combination of all items") { FF.`fold should combine all items`(GEN, EQ) },
+      Law("Foldable Laws: combineAll is an alias for fold") { FF.`combineAll consistent with fold (alias)`(GEN, EQ) },
+      Law("Foldable Laws: reduceLeftToOption combines all items into an optional value") { FF.`reduceLeftToOption returns Option value`(GENB, EQOptionInt) },
+      Law("Foldable Laws: reduceRightToOption combines all items into an optional value") { FF.`reduceRightToOption returns Option value`(GENB, EQOptionInt) },
+      Law("Foldable Laws: reduceLeftOption consistent with reduceLeftToOption") { FF.`reduceLeftOption returns Option value`(GEN, EQOptionInt) },
+      Law("Foldable Laws: reduceRightOption consistent with reduceRightToOption") { FF.`reduceRightOption returns Option value`(GEN, EQOptionInt) }
+    )
+  }
+
+  fun <F> laws(FF: Foldable<F>, GA: Applicative<F>, GENK: GenK<F>, EQK: EqK<F>): List<Law> {
+    val GEN: Gen<Kind<F, Int>> = GENK.genK(Gen.intSmall())
+
+    val EQKInt: Eq<Kind<F, Int>> = EQK.liftEq(Int.eq())
+    val EQKUnit = EQK.liftEq(Eq.any())
+
+    return listOf(
+      Law("Foldable Laws: orEmpty consistent with just empty") { FF.`orEmpty consistent with just empty`(GA, EQKInt) },
+      Law("Foldable Laws: traverse_ consistent with foldRight") { FF.`traverse_ consistent with foldRight`(GA, GENK, GEN, EQKUnit) },
+      Law("Foldable Laws: sequence_ consistent with traverse_") { FF.`sequence_ consistent with traverse_`(GA, GENK, EQKUnit) }
```

Comment:
I am against adding the applicative to the params. `traverse_` doesn't care about what instance is used, so just choose one for the tests. `Id` `Option` etc all are decent choices.

Reply:
I've pushed it to ask because I didn't know how to proceed there!! thanks for your fast response!! :D